hardware_interface/include/hardware_interface/robot_hw.h

```diff
@@ -148,33 +168,66 @@ class RobotHW : public InterfaceManager
     ERROR
   };
 
-  // Return (in realtime) the state of the last doSwitch()
+  /** \brief Return (in realtime) the state of the last doSwitch() */
   virtual SwitchState switchResult() const
   {
     return DONE;
   }
 
-  // Return (in realtime) the state of the last doSwitch() for a given controller
+  /** \brief Return (in realtime) the state of the last doSwitch() for a given controller */
   virtual SwitchState switchResult(const ControllerInfo& /*controller*/) const
   {
     return DONE;
   }
+  /**\}*/
 
-  /**
-   * Reads data from the robot HW
+  
+  /** \name Control Loop
+   * 
+   */
+   /**\{*/
+
+  /** \brief Read data from the robot hardware.
+   *
+   * The read method is part of the control loop cycle (\ref read, update, \ref write) 
+   * and is used to populate the robot state from the robot's hardware resources
+   * (joints, sensors, actuators). This method should be called before 
+   * controller_manager::ControllerManager::update() and \ref write.
+   * 
+   * \note The name \ref read refers to reading state from the hardware.
+   * This complements \ref write, which refers to writing commands to the hardware.
+   *
+   * Querying WallTime inside \ref read is not realtime safe. The input parameters
+   * \ref time and \ref period make it possible to inject time from any 
+   * realtime backend framework.
```

Comment:
I've added a reason why the `read` and `write` methods take `time` and `duration` parameters. I hope this is their intended use. Please let me know if not.

Reply:
Yep, that's their intended use. Good addition.

A bit of a nitpick, but "input parameters" is a bit redundant, "parameters" is fine. I would also reword it to "inject time from a realtime source", but that might just be my personal writing preferences creeping in ;)

Reply:
Sounds nice :) Let's do it that way.